### PR TITLE
Pkg 4499 pypdf 4.2.0

### PR DESCRIPTION
**Pull request by Jrice1317**
pypdf 4.2.0 ❄️ 

**Destination channel:** {defaults}

### Links

- [PKG-4499](https://anaconda.atlassian.net/browse/PKG-4499) 
- [Upstream repository](https://github.com/py-pdf/pypdf/tree/4.2.0)
- [Upstream changelog/diff](https://github.com/py-pdf/pypdf/blob/4.2.0/CHANGELOG.md)

### Explanation of changes:
- Remove noarch

[PKG-4499]: https://anaconda.atlassian.net/browse/PKG-4499?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -48,64 +37,55 @@ outputs:
...
   - name: pypdf-with-crypto
-    build: &noarch-generic
-      noarch: generic
+    build:
+      skip: True  # [py<36]
     requirements:
+      host:
+        - python
+        - flit-core >=3.9,<4
+        - pip
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
Because it won't "build", we don't need these dependencies.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -48,64 +37,55 @@ outputs:
...
 
   - name: pypdf-with-image
-    build: *noarch-generic
+    build:
+      skip: True  # [py<36]
     requirements:
+      host:
+        - python
+        - flit-core >=3.9,<4
+        - pip
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
Same

File: recipe/meta.yaml [hunk 3]
```diff
@@ -48,64 +37,55 @@ outputs:
...
 
   - name: pypdf-with-full
-    build: *noarch-generic
+    build:
+      skip: True  # [py<36]
     requirements:
+      host:
+        - python
+        - flit-core >=3.9,<4
+        - pip
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
And same.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Good job @Jrice1317! It wasn't easy but you made it!

**Review by psteyer — APPROVED** [review]

**Pull request merged**